torch/_inductor/codegen/triton.py

```diff
@@ -722,65 +723,56 @@ def indexing(
         index = self.simplify_indexing(index)
         index_vars = index.free_symbols
         index_str = texpr(self.rename_indexing(self.codegen_indexing(index)))
-        indirect_indexing = self.is_indirect_indexing(index)
+
+        mask_vars: Set[str] = set()
+        for var in index_vars:
+            if var.name.startswith("tmp"):
+                # indirect indexing
+                cse_var = self.cse.varname_map[var.name]
+                mask_vars.update(cse_var.mask_vars)
+            else:
+                # var is one of xN, yN or rN
+                assert var.name[0] in "xyr", var.name
+                mask_vars.add(f"{var.name[0]}mask")
 
         need_dense = (
             config.triton.dense_indexing
             or dense_indexing
-            or indirect_indexing
             or self._load_mask is not None
         ) and index != 0
 
         have_dense = True
         have_loop_vars = False
-        mask = []
-        dense_mask = []
+        dense_mask = set()
```

Comment:
Python sets have nondeterministic order.  If we use them we will need to call sort later.